jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -139,6 +140,8 @@ public HttpConnection(HttpConfiguration configuration, Connector connector, EndP
         _httpChannel = newHttpChannel(connector.getServer(), configuration);
         _requestHandler = newRequestHandler();
         _parser = newHttpParser(configuration.getHttpCompliance());
+        _minBufferSpace = configuration.getMinInputBufferSpace() < 0 ? Math.min(1024, configuration.getInputBufferSize()) : configuration.getMinInputBufferSpace();
```

Comment:
I think it should be a size not a percentage.  The value of the effort to do a read does not scale with the size of the total buffer.

However, I could be convinced to leave out the configuration from HttpConfiguration for now.... my only concern is that does not give us a method to turn it off if it proves to be a problem for some applications?

Reply:
> I think it should be a size not a percentage. The value of the effort to do a read does not scale with the size of the total buffer.

Not sure I follow?
This work is about reusing the input buffer when it's retained, so it's about memory, not reads -- the number of reads is the same.
Since it's about memory, if an application configures buffers of 256 KiB because it knows it has large content to read, isn't a fixed value wrong as it would force a large read into a very small remaining? And the same for smaller input buffer sizes: the fixed value would make it so the buffer will never be reused.

With a percentage, we give users 1 knob, and if it is really necessary we can add `minInputBufferSpace` later.
With a fixed size, it will only work well when the fixed size is the "right" percentage of the input buffer size.

Reply:
I picked 1024 because it is the same order of magnitude as the MTU of 1500 for many networks.   Probably better to actually have 1500, so enough space to read a whole packet without fragmenting it.

If a 256 KB input buffer is specified, then I think it would be silly to dispatch to the application with 32KB of unused buffer!